### ISSUE TITLE
`pℤₚ` has index `p` in `ℤₚ`

### ISSUE THREAD

**Issue by YaelDillies** [ImperialCollegeLondon/FLT#279]
The index of $p\mathbb Z_p$ inside $\mathbb Z_p$ is $p$. The precise version I have stated is that the index of any compact $\mathbb Z_p$-submodule of $\mathbb Q_p$ has index $p$ relative to $s$.

**Comment by mpenciak** [comment]
I was looking at the statement of the theorem referencing this here: https://github.com/ImperialCollegeLondon/FLT/blob/c51f2573c22ded973f354f56f598507b0f349f51/FLT/Mathlib/NumberTheory/Padics/PadicIntegers.lean#L58-L62

and I think (unless I'm misunderstanding things, which is totally possible) this isn't quite stated correctly.

In particular, if we take `s = ℚ_[p]` itself (where `x` is invertible) then `x • ℚ_[p]` will be `ℚ_[p]` again and the index is 1. (not  `‖x‖₊⁻¹` in general).

I think this is mostly alright because the only place where it's being used so far is in the `distribHaarChar_padic_padicInt` lemma where `s` is `ℤ_[p]` itself (if I'm understanding the coercions correctly).

I may be totally off, but I wanted to bring this up as a possible issue.

**Comment by YaelDillies** [comment]
Good point! We forgot to assume that `s` is compact. Statement fixed in #407.

**Comment by Aaron1011** [comment]
A special case of this was added in https://github.com/ImperialCollegeLondon/FLT/pull/408

**Comment by YaelDillies** [comment]
Great! This task is done by me.
